### PR TITLE
New Package: citations-0.5.1

### PR DESCRIPTION
**Pull request by misuchiru03**
#### Testing the changes
- I tested the changes in this PR: **YES**

#### New package
- This new package conforms to the [package requirements](https://github.com/void-linux/void-packages/blob/master/CONTRIBUTING.md#package-requirements): **YES**

#### Local build testing
- I built this PR locally for my native architecture, (x86_64-GLIBC)
- aarch64-musl
- aarch64
- armv7l

### CODE REVIEW

**Comment by misuchiru03** [comment]
I do not think it likes being cross-compiled. I do not have an armv7l or aarch64 machine to verify native builds.

**Comment by classabbyamp** [comment]
add `build_helper=rust`, at minimum. you may need rust-std in makedepends

**Comment by misuchiru03** [comment]
If the package is built natively, it is fine as is. There is a `build.ninja` that expects the file to build natively and the cross build places the built files in a different location so cross builds fail. I have added a check for Cross Build only to make sure it finds the file properly and cross builds built fine on my system.

File: srcpkgs/citations/template [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+license="GPL-3.0-or-later"
+homepage="https://gitlab.gnome.org/World/citations"
+distfiles="https://gitlab.gnome.org/World/citations/-/archive/${version}/citations-${version}.tar.gz"
+checksum=77ac6a7cc442de74eccfc3f07687803cd1feca2da348753eddd34404b94bf253
+_meson_builddir="${XBPS_BUILDDIR}/citations-${version}/build"
+
+# Allows the cross build to know where the file is placed--build.ninja does not account for cross build
+pre_build() {
+	if [ "${CROSS_BUILD}" ]; then
+		sed -i 's, && cp src/release/citations src/citations,,' ${_meson_builddir}/build.ninja
```
Comment (abenson) [inline srcpkgs/citations/template:22]: Use `vsed`

**Pull request merged**